cmd/registry/root.go

```diff
@@ -0,0 +1,259 @@
+	logKeyNotExists(exists, "providers", dagJson)
+	for _, provider := range providers {
+		providerJson := provider.(map[string]interface{})
+		providerId, nexists := providerJson["name"].(string) // displayName??
+		logKeyNotExists(nexists, "name", providerJson)
+		log.Infof("Adding provider required for DAG: %s", providerId)
+		addProviderByName(providerId)
+	}
+}
+
+func getDagRoute(dagName string, dagVersion string) string {
+	filledDagRoute := fmt.Sprintf(dagRoute, dagName, dagVersion)
+	getDagUrl := url.URL{
+		Scheme: "https",
+		Host:   registryHost,
+		Path:   fmt.Sprintf("%s/%s", registryApi, filledDagRoute),
+	}
+	return getDagUrl.String()
+}
+
+func downloadDagToFile(sourceUrl string, path string) {
+	file, err := create(path)
+	client := http.Client{
+		CheckRedirect: func(r *http.Request, via []*http.Request) error {
+			r.URL.Opaque = r.URL.Path
+			return nil
+		},
+	}
+	resp, err := client.Get(sourceUrl)
+	logFatal(err)
+	//goland:noinspection GoUnhandledErrorResult
+	defer resp.Body.Close()
+	_, err = io.Copy(file, resp.Body)
+	//goland:noinspection GoUnhandledErrorResult
+	defer file.Close()
+	log.Infof("Downloaded file %s from %s", path, sourceUrl)
+}
+
+func newRegistryCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "registry",
+		Aliases: []string{"r"},
+		Short:   "Utilize the Astronomer Registry",
+	}
+	cmd.AddCommand(
+		newRegistryAddCmd(),
+		newRegistryInitCmd(),
+	)
+	return cmd
+}
+
+func newRegistryInitCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "init",
+		//Aliases: []string{"r"},
+		Short: "Utilize the Astronomer Registry",
+		Long:  "Utilize the Astronomer Registry to initialize a project from a template.",
+	}
+	//cmd.AddCommand(
+	//	newRegistryAddDagCmd(),
+	//	newRegistryAddProviderCmd(),
+	//)
+	return cmd
+}
+
+func newRegistryAddCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "add",
+		Aliases: []string{"a"},
+	}
+	cmd.AddCommand(
+		newRegistryAddDagCmd(),
+		newRegistryAddProviderCmd(),
+	)
+	return cmd
+}
+
+func newRegistryAddDagCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "dag [DAG ID]",
+		Aliases: []string{"d"},
+		Short:   "Utilize the Astronomer Registry to add a DAG",
+		Long:    "Utilize the Astronomer Registry to download a DAG to your local project.",
+		Run: func(cmd *cobra.Command, args []string) {
+			var dagName string
+
+			// if a dagName was provided in the args we use it
+			if len(args) > 0 {
+				dagName = args[0]
+			} else {
+				// no dagName was provided so ask the user for it
+				dagName = input.Text("Enter DAG ID to download: ")
+			}
+			downloadDag(dagName, dagVersion)
+		},
+	}
+	cmd.Flags().StringVar(&dagVersion, "version", "latest", "Optional DAG Version to Download.")
+	return cmd
+}
+
+func newRegistryAddProviderCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "provider [PROVIDER]",
+		//Aliases: []string{"p"},
+		Short: "Utilize the Astronomer Registry to add a Provider",
```

Comment:
```suggestion
		Short: "Download a provider package from the Astronomer Registry",
```
For my understanding: how do we anticipate someone to use this command? How is this easier than just pasting into `requirements.txt`?

Reply:
@jwitz - Plenty of new to airflow folks. I think it's *absolutely* easier to type `astro registry provider add snowflake` than go look up what the correct import and version is, know what `requirements.txt` is, add it, version it - etc. ALSO this lets you `astro registry dag add ...` and have it auto-add the correct provider 😎